pkg/csi/service/vanilla/controller.go

```diff
@@ -906,6 +921,464 @@ func (c *controller) getAccessibleTopologiesForDatastore(ctx context.Context, vc
+			"failed to club accessibility requirements by VC. Error: %+v", err)
+	}
+	log.Debugf("Topology accessibility requirements per VC are %+v", vcTopologySegmentsMap)
+
+	if !volTaskAlreadyRegistered {
+		// Check if topology domains have been provided in the vSphere CSI config secret.
+		// NOTE: We do not support kubernetes.io/hostname as a topology label.
+		if c.managers.CnsConfig.Labels.TopologyCategories == "" && c.managers.CnsConfig.Labels.Zone == "" &&
+			c.managers.CnsConfig.Labels.Region == "" {
+			return nil, csifault.CSIInvalidArgumentFault, logger.LogNewErrorCode(log, codes.InvalidArgument,
+				"topology category names not specified in the vsphere config secret")
+		}
+
+		// Iterate through each VC and its accessibility requirements to try and create a volume.
+		// If it fails for any reason, move unto the next VC in list.
+		var topologySegmentsList []map[string]string
+		for vcHost, topologySegmentsList = range vcTopologySegmentsMap {
+			// Get VC instance.
+			vcenter, err = common.GetVCenterFromVCHost(ctx, c.managers.VcenterManager, vcHost)
+			if err != nil {
+				return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
+					"failed to get vCenter instance for host %q. Error: %+v", vcHost, err)
+			}
+
+			// If Storage policy is given, check if it exists in the VC. If not found, continue to next VC.
+			var storagePolicyID string
+			if scParams.StoragePolicyName != "" {
+				storagePolicyID, err = vcenter.GetStoragePolicyIDByName(ctx, scParams.StoragePolicyName)
+				if err != nil {
+					// TODO: Multi-VC: Catch NotFound error and continue only on NotFound error.
+					log.Warnf("failed to get storage policy ID for name %q in VC %q",
+						scParams.StoragePolicyName, vcHost)
+					continue
+				}
+				log.Infof("Found ID %q for storage policy name %q", storagePolicyID,
+					scParams.StoragePolicyName)
+			}
+
+			// Get shared accessible datastores for topology segments associated with the vcHost.
+			sharedDatastores, err = placementengine.GetSharedDatastores(ctx,
+				placementengine.VanillaSharedDatastoresParams{
+					Vcenter:              vcenter,
+					TopologySegmentsList: topologySegmentsList,
+					StoragePolicyID:      storagePolicyID,
+					IsTopologyPreferentialDatastoresFSSEnabled: isTopologyPreferentialDatastoresFSSEnabled,
+					IsCSINodeIdFeatureEnabled:                  isCSINodeIdFeatureEnabled,
+				})
+			if err != nil {
+				return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
+					"failed to get shared datastores for topology segments %+v in vCenter %q. Error: %+v",
+					topologySegmentsList, vcHost, err)
+			}
+			if len(sharedDatastores) == 0 {
+				log.Warnf("no shared datastores found for accessibility requirements pertaining to vCenter %q",
+					vcHost)
+				continue
+			}
+			// Filter datastores based on user access.
+			if isAuthCheckFSSEnabled {
```

Comment:
isAuthCheckFSSEnabled is not required, as in multi-vcetner feature this is always enabled.

Reply:
Done